cumulusci/tasks/bulkdata.py

```diff
@@ -797,6 +799,313 @@ def _sqlite_dump(self):
+                        self._is_any_custom_api_name(field["name"]),
+                        self._is_core_field(field["name"]),
+                        self._is_required_field(field),
+                        self._is_lookup_to_included_object(field),
+                    ]
+                ):
+                    if self._is_field_mappable(obj, field):
+                        self.schema[obj][field["name"]] = field
+
+                        if field["type"] == "reference":
+                            for target in field["referenceTo"]:
+                                if (
+                                    obj in self.mapping_objects
+                                    and target in self.mapping_objects
+                                ):
+                                    if obj not in self.refs:
+                                        self.refs[obj] = set()
+                                    self.refs[obj].add(target)
+
+    def _build_mapping(self):
+        objs = set(self.schema.keys())
+        stack = self._split_dependencies(objs, self.refs)
+
+        field_sort = (
+            lambda f: "  " + f
+            if f == "Id"
+            else (" " + f if f in self.core_fields else f)
+        )
+
+        self.mapping = OrderedDict()
+        for obj in stack:
+            key = "Insert {}".format(obj)
+            self.mapping[key] = OrderedDict()
+            self.mapping[key]["sf_object"] = "{}".format(obj)
+            self.mapping[key]["table"] = "{}".format(obj.lower())
+            fields = []
+            lookups = []
+            for field in self.schema[obj].values():
+                if field["type"] == "reference":
+                    lookups.append(field["name"])
+                else:
+                    fields.append(field["name"])
+            self.mapping[key]["fields"] = OrderedDict()
+            if fields:
+                if "Id" not in fields:
+                    fields.append("Id")
+                fields.sort(key=field_sort)
+                for field in fields:
+                    self.mapping[key]["fields"][field] = (
+                        field.lower() if field != "Id" else "sf_id"
+                    )
+            if lookups:
+                lookups.sort(key=field_sort)
+                self.mapping[key]["lookups"] = OrderedDict()
+                for field in lookups:
+                    self.mapping[key]["lookups"][field] = {
+                        "table": self.schema[obj][field]["referenceTo"][0].lower()
+                    }
+                    if len(self.schema[obj][field]["referenceTo"]) > 1:
+                        self.logger.warning(
+                            "Field {}.{} is a polymorphic lookup, which is not supported".format(
+                                obj, field
+                            )
+                        )
+
+    def _run_task(self):
+        self._collect_objects()
+        self._build_schema()
+        self._build_mapping()
```

Comment:
I'd suggest organizing this class with _run_task first, then the 3 methods called here, followed by other methods that they end up calling. It makes the module easier to read and understand in one pass. I also wouldn't complain if you feel inclined to add some docstrings.